openquake/hazardlib/gsim/yu_2013.py

```diff
@@ -0,0 +1,242 @@
+    term3 = a2cd*np.exp(a2ce*mag)
+    return np.exp((term1 - term2) / a2cc) - term3
+
+
+def fnc(ra, *args):
+    """
+    Find the rupture to site distance
+    """
+    #
+    # epicentral distance
+    repi = args[0]
+    #
+    # azimuth
+    theta = args[1]
+    #
+    # magnitude
+    mag = args[2]
+    #
+    # coefficients
+    coeff = args[3]
+    #
+    #
+    rb = rbf(ra, coeff, mag)
+    t1 = ra**2 * (np.sin(np.radians(theta)))**2
+    t2 = rb**2 * (np.cos(np.radians(theta)))**2
+    xx = ra * rb / (t1+t2)**0.5
+    return abs(repi-xx)
+
+
+class YuEtAl2013Ms(GMPE):
+    """
+    Implements the Yu et al. (2013) GMPE used for the calculation of the 2015
+    version of the national seismic hazard maps for China. Note that magnitude
+    supported is Ms.
+    """
+
+    #: Supported tectonic region type is active shallow crust
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Supported intensity measure types are peak ground velocity and
+    #: peak ground acceleration
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA,
+        PGV,
+        SA
+    ])
+
+    #: Supported intensity measure component is geometric mean (supposed)
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    #: Supported standard deviation types is total
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL
+    ])
+
+    #: No site parameters required
+    REQUIRES_SITES_PARAMETERS = set(())
+
+    #: Required rupture parameter is magnitude
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag',))
+
+    #: Required distance measures are epicentral distance and azimuth
+    REQUIRES_DISTANCES = set(('repi', 'azimuth'))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+            """
+            See :meth:`superclass method
+            <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+            for spec of input and result values.
+            """
+            # Check that the requested standard deviation type is available
+            assert all(stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+                       for stddev_type in stddev_types)
+            #
+            # Set coefficients
+            C = self.COEFFS[imt]
+            if rup.mag > 6.5:
+                a1ca = C['ua']
+                a1cb = C['ub']
+                a1cc = C['uc']
+                a1cd = C['ud']
+                a1ce = C['ue']
+                a2ca = C['ia']
+                a2cb = C['ib']
+                a2cc = C['ic']
+                a2cd = C['id']
+                a2ce = C['ie']
+            else:
+                a1ca = C['a']
+                a1cb = C['b']
+                a1cc = C['c']
+                a1cd = C['d']
+                a1ce = C['e']
+                a2ca = C['ma']
+                a2cb = C['mb']
+                a2cc = C['mc']
+                a2cd = C['md']
+                a2ce = C['me']
+            #
```

Comment:
Is this the same as the function define before? Could they be unified?

Reply:
Done